url27db8982cd_leohinterlang_argFace_tgz-pJ8-com_fidelis_argface_ArgSampleJ8/src/com/fidelis/argface/sample/Sample.java

```diff
@@ -0,0 +1,149 @@
+/**
+ *+
+ *	Sample.java
+ *-
+ */
+package com.fidelis.argface.sample;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Sample program.
+ * 
+ * @version 1.0.0
+ * @author Leo Hinterlang
+ *
+ */
+public class Sample {
+	private SampleCLI cli;
+	private String [] lines = new String[6];
+	private int lineNumber;
+	private int lineStartIndex;
+	private int lineNextIndex;
+
+	/**
+	 * Sample program to illustrate the use of the ArgFace CLI.
+	 * Here is the usage specification:
+	 * <blockquote><code>
+	 * sample [-a] [-b/--brand] [-c &lt;name&gt] [-d/--dir &lt;path&gt;]
+	 *   [find &lt;pattern>] file...
+	 * </code></blockquote>
+	 * 
+	 * @param args command line arguments
+	 */
+	public static void main (String[] args) {
+		
+		// Create this program class.
+		Sample prog = new Sample();
+		
+		// Execute the program.
+		if (! prog.execute(args)) {
+			System.out.println("Error exit");
+			System.exit(1);
+		}
+	}
+	
+	private boolean execute (String [] args) {
+		
+		// SampleCLI is a POJO that contains all of the logic for
+		// the command line interface.
+		cli = new SampleCLI();
+		
+		// Parse the command line arguments.
+		if (! cli.parse(args)) {
+			return false;
+		}
+		// Display the CLI results.
+		System.out.println("Results ----");
+		System.out.println("Options:");
+		cli.report("options");
+		System.out.println("Operands:");
+		cli.report("operands");
+		return true;
+	}
+	
+	private void processFile (String filename) {
+		String pattern = "public";
+		if (cli.isFindOperand()) {
+			pattern = cli.getPatternOperand();
+		}
+		Path file = null;
+		if (cli.isDirOption()) {
+			file = Paths.get(cli.getDirPath() + "/" + filename);
+		} else {
+			file = Paths.get(filename);
+		}
+		Charset charset = Charset.forName("US-ASCII");
+		try {
+			BufferedReader reader = Files.newBufferedReader(file, charset);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *reader*. Execution paths that do not contain closure statements were detected. Ensure *reader* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).